### DISCUSSION TITLE
Multilingual support

### DISCUSSION THREAD

**Discussion by cwhauser** [eclipse-birt/birt#1295] [Q&A] [answered]
Needing to add multilingual support to an existing BIRT report.  Also known as localization.  
Any assistance would be greatly appreciated.

**Comment by speckyspooky** [comment, marked as answer]
The first step is you need a standard property file by default you need for 2 languages 3 files:
file 01 - default language: <filename>.properties
file 02 - language DE: <filename>_de.properties
file 03 - language EN: <filename>_en.properties

The postfix "_de.properties" is a standard Java mechanism so you would be able to use the long version of "_de_DE.properties"
because German, English and other languages could have subcategories of languages.
Be aware the postfix use the ISO-language code and not the ISO-country codes 2 examples:
- czech republic: ISO-language code "cs" but country-code "cz"
- Sweden: ISO-language code "sv" but country-code "se"

Conten, structure like <key>=<label>, e.g.
label1=Ich bin eine Beschriftung
example=Beispiel

The next step is to add the file 01 (default language) to your report ressources

![add_file-580x197](https://github.com/eclipse-birt/birt/assets/41593722/fa3289c3-4961-4578-8392-ef5981da5583)

If you have added the file then you use the properties on BIRT-labels on parameters:

![2023-05-27 08_29_57-workspace - C__Users_gutma_Downloads_rptDesign_XMLData_b2beaccountpdf-aulatepaym](https://github.com/eclipse-birt/birt/assets/41593722/625d7448-bc61-4645-8b4b-e89dd06705eb)

I hope it gives you the first steps and ideas for your next steps.

**Reply by cwhauser** [reply]
Thank you VERY MUCH for the feedback.  I've never used a properties file so I've got some things to learn in order to apply what you've suggested.  Plus the multilingual requirement has been put on hold for now.  So I have some breathing room - time to research more and try your suggestions. 
Thanks again!
